# T1.5:selection of subset of relevant companies

<span style="font-size:20px;">  The goal is  to take a subset of relevant companies where all the selected companies will have at least one review and no missing values in our new subset.
Below is the code where we clean the data and we extract a new file called "clean_data_set.csv"

In [104]:
#import the needed librires
import pandas as pd

##Set display optoins
pd.set_option("display.max_columns", 85)
pd.set_option("display.max_rows", 85)

#read the data set from the csv file
df= pd.read_csv('data/TechLabsDataset.csv',index_col=0)
##Drop Nan values from the data set
##reset index after drop
df_clean_data=df.dropna(axis=0).reset_index(drop=True)
#Adding a new column where we count the number of reviews for each company.
df_clean_data['number_of_reviews']=(df_clean_data.groupby('name')['review_EN'].transform('count'))
##In our new dataset without NaN values how many companies do we have in total?
#Get unique values using set
df_clean_Comp=set(df_clean_data.name)
print(df_clean_Comp)
print('\n Number of companies ', len(df_clean_Comp))

#create a new csv file with the clean data into the Data directory
#with the command "index=False", we prevent a new index column to be added to the extracted file

df_clean_data.to_csv('data/clean_data_set.csv',index=False)

{'ITS Breda BV', 'Weiner Logistics Bv', 'Serpenti Logistics BV', 'Dekkers Transport Holland', 'Otentic Logistics B.V.', 'JTS Logistics BV', 'Id logistics b.v.', 'Noach Logistics Koeltransport en vriestransport', 'Vortex Logistics', 'TCI International Logistics B.V.', 'BK Logistics B.V.', 'Neovia Logistics Services, LLC', 'Auction Logistics B.V.', 'Arma Logistics B.V.', 'Cityline Logistics BV', 'Slimm Logistics Bv', 'Nord Cargo BV', 'Kennis Transport en Logistics BV', 'LV Logistics (ontstaan uit J. Lensveld & Zonen BV)', 'Soetens Warehousing & Logistics', '4 Logistics Netherlands B.V.', 'Rhenus Contract Logistics Eindhoven B.V.', 'Lucas Logistics B.V.', 'ND Logistics', 'Elst Transport', 'Bos Logistics Schiphol', 'Jan de Rijk Logistics HQ', '24/7 Logistic Solutions B.V.', 'Alp Transport en Logistics', 'Lankveld Logistics B.V.', 'Stella Container Logistics BV', 'RMI Global Logistic Services', 'Crane Worldwide Logistics', 'Hartog Logistics B.V.', 'ATL Logistics BV', '010 Logistics', 'WDeli

<span style="font-size:20px;">  Import and read the new csv file


In [102]:
df_sub=pd.read_csv('data/clean_data_set.csv')
df_sub

name      type_company        lat       lon  \
0         Dekkers Transport Holland  Trucking company  51.597249  5.027990   
1         Dekkers Transport Holland  Trucking company  51.597249  5.027990   
2         Dekkers Transport Holland  Trucking company  51.597249  5.027990   
3         Dekkers Transport Holland  Trucking company  51.597249  5.027990   
4         Dekkers Transport Holland  Trucking company  51.597249  5.027990   
...                             ...               ...        ...       ...   
3393                 L.H. Logistics  Trucking company  51.617957  4.736123   
3394                 L.H. Logistics  Trucking company  51.617957  4.736123   
3395                 L.H. Logistics  Trucking company  51.617957  4.736123   
3396                 L.H. Logistics  Trucking company  51.617957  4.736123   
3397  InTime Logistics & Consulting  Trucking company  52.049578  5.117079   

            city  avg_stars_num  n_reviews_num encoded_user  local_guide  \
0        Tilburg            4.6           53.0        Pp1)c          0.0   
1        Tilburg            4.6           53.0        *nk1l          0.0   
2        Tilburg            4.6           53.0        69lOZ          0.0   
3        Tilburg            4.6           53.0        &7jv4          0.0   
4        Tilburg            4.6           53.0        K2#6M          1.0   
...          ...            ...            ...          ...          ...   
3393       Breda            3.5            8.0        AMDIc          0.0   
3394       Breda            3.5            8.0        ivODx          0.0   
3395       Breda            3.5            8.0        K1S^G          0.0   
3396       Breda            3.5            8.0        RP^Ug          1.0   
3397  Nieuwegein            4.7            3.0        wqc@z          1.0   

      clean_other_review_num published_date  today_date  stars_num  \
0                        5.0     a year ago  2022-09-24        5.0   
1                        2.0     a year ago  2022-09-24        5.0   
2                        3.0    4 years ago  2022-09-24        5.0   
3                       27.0   3 months ago  2022-09-24        5.0   
4                       24.0   4 months ago  2022-09-24        5.0   
...                      ...            ...         ...        ...   
3393                     1.0   9 months ago  2022-09-24        1.0   
3394                     2.0     a year ago  2022-09-24        1.0   
3395                    17.0     a year ago  2022-09-24        5.0   
3396                   101.0     a year ago  2022-09-24        5.0   
3397                    13.0   9 months ago  2022-09-22        5.0   

                                              review_EN original_lang  \
0     Best employer ever. Worked there for approxima...           ENG   
1     better place with very good people everything ...           ENG   
2                                                  Nice           ENG   
3      Good firm. The staff are very welcoming and f...       NOT_ENG   
4       Top company, and coffee, so recommended\r\n\r\n       NOT_ENG   
...                                                 ...           ...   
3393   Untrustworthy people, they are good at lying....       NOT_ENG   
3394   Drives antisocial and overtakes right on the ...       NOT_ENG   
3395   In a word, toppers! Always think in possibili...       NOT_ENG   
3396                              First service\r\n\r\n       NOT_ENG   
3397                                     Unique\r\n\r\n       NOT_ENG   

      number_of_reviews  
0                    19  
1                    19  
2                    19  
3                    19  
4                    19  
...                 ...  
3393                  6  
3394                  6  
3395                  6  
3396                  6  
3397                  1  

[3398 rows x 16 columns]

<span style="font-size:20px;"> ***- The new file we created has 16 columns and 3398 records.***

<span style="font-size:20px;"> ***- The relevant companies with at least 1 review is in total 184***

<span style="font-size:20px;"> Below is the companies sorted by the number of reviews that each one has

In [103]:
print(df_sub.groupby("name")['number_of_reviews'].size().sort_values(ascending=False))

name
First Logistics B.V.             293
Van Spreuwel Transport           288
NUNNER Logistics BV              143
Claassen Logistics               135
ID Logistics Benelux             123
                                ... 
Id logistics b.v.                  1
Global Logistics Holland B.V.      1
VDR logistics                      1
Frankena Logistics                 1
Über Logistics                     1
Name: number_of_reviews, Length: 184, dtype: int64



<span style="font-size:20px;">***Define critiria for companies selection***

<span style="font-size:20px;">1.Selection of companies that have all the data complete.
(this has already been achieved by the data cleansing process)

<span style="font-size:20px;">2.Selection of companies that have more than several reviews that we still have to define, depending on the first results of  WP2 and WP3.


<span style="font-size:20px;">The following questions arise.
Is 184 companies a sufficient sample or do we need a smaller one with more reviews than one per company?
maybe companies with more than 10 or 20 reviews?

<span style="font-size:20px;">Does the project implementation time allow us to collect sufficient data from new sources and complete the processing?